### ISSUE TITLE
Accessibility: Focus gets recycled

### ISSUE THREAD

**Issue by getflourish** [Akryum/vue-virtual-scroller#499]
This is a demo using `RecycleScroller`.

The selection highlight is visible when the item is focussed. When scrolling fast enough, you can see that other items "look" focussed although they should not be. I assume that the view that was initially focussed gets recycled as is, including the focus. 

## Problem
While scrolling, the selection highlight randomly shows up for items that aren’t actually selected. In the demo I select the item at `index = 0` but items with other indices show up as being selected. At some point, the selection gets lost completely. 

![Kapture 2020-09-03 at 15 55 56](https://user-images.githubusercontent.com/183989/92124358-f7e96b80-edfd-11ea-99c2-f1b5051f8888.gif)

**Comment by ChristianKienle** [comment]
I have also run into this issue. But there is more to it I think. Keeping track of the focused index is only one out of many possible things that could/should(?) be tracked - right? What with selected text in an item? If you select something and then scroll far enough you will see that the selected text is messed up.

When I realised that I concluded for myself that this and similar things should not be part of this component...

I then recalled how this is handled in other places. Long lists on iOS for example are implemented pretty generically. On iOS the developer has to "reset" every cell/item before it is being reused. I think this would be good enough even for this component. However I think this is currently not possible. 

Another edge case that supports the need for some kind of generic solution is this:

Imagine a list of input elements. The user now starts editing an input element and then scrolls. At some point this element will be re-used. What should happen in this case? This is highly app specific. Some apps might discard the current value of the input - others might commit the current value…

Does this make sense?

**Comment by getflourish** [comment]
I agree that this issue is just one example of many and I don’t expect this specific component to get everything right. I think if one treats it as a better way to *display* many items, it works great pretty well. But I can imagine that supporting all sort of *interactive* and *accessibility* features may require a lot of work, maybe even a rewrite.

Just for the sake of inspiration, here’s an implementation used by the Github Desktop app. https://github.com/desktop/desktop/blob/development/app/src/ui/lib/list/list.tsx 

It seems to build on top of `react-virtualized` and it may be an example of what you describe: implementation specific details added on top. The React library does seem to manage focus: https://bvaughn.github.io/react-virtualized/#/components/ArrowKeyStepper

**Comment by ChristianKienle** [comment]
@getflourish So what I like about Vue is that they made the hard decisions for me. :D Hard decisions like what to include/support out of the box and what to leave to the individual developer.

Reading your response made me think and reconsider… Maybe a component like this would be so much better if there were some key things already included and others not: Decisions are required. :D

And thinking more about this: Focusing is also already built into the corresponding "component" in UIKit/iOS…. As is selecting things…

I am currently trying to make vue-virtual-scroller a bit more "usable" for my own UI component library since I use it a lot. Let's see how this goes. :D

**Comment by getflourish** [comment]
@ChristianKienle I also wish there was "the one and only way" and it would just work out of the box. As you say, Vue is rather opinionated and requires little to no choice for common requirements (think: Router, Vuex, Scoped Styles) compared to the dozens of implementations for React. Ideally there would be an official way but this is the chance we have: to contribute an implementation or at least a list of requirements :) I hope to share my hacks soon. Maybe you'll also figure some things out :)
